server/cluster/cluster.go

```diff
@@ -605,26 +604,31 @@ func (c *RaftCluster) runReplicationMode() {
 // Stop stops the cluster.
 func (c *RaftCluster) Stop() {
 	c.Lock()
-	if !c.running.CompareAndSwap(true, false) {
+	if !c.running {
 		c.Unlock()
 		return
 	}
-
 	c.coordinator.stop()
 	c.cancel()
+	c.running = false
```

Comment:
Previously we will first set the running to false, then stop the coordinator. Maybe we can keep the same, see https://github.com/tikv/pd/pull/4927.

Reply:
I can change, but why it matters if c.running now is protected by Lock() which prevent all other reads with RLock()?

Reply:
Maybe it doesn't matter,  but we have used that logic for a long time, and it could be safer.